static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js

```diff
@@ -1,129 +1,126 @@
 // @flow
 
-import qwery from 'qwery';
 import fastdom from 'fastdom';
 import ophan from 'ophan/ng';
 import userAccount from 'common/modules/navigation/user-account';
 
-const html = qwery('html')[0];
-const menuItems = qwery('.js-close-nav-list');
+const sidebar = document.getElementById('main-menu');
 const enhanced = {};
 
-const weShouldEnhance = (checkbox: HTMLInputElement): ?boolean =>
-    checkbox && !enhanced[checkbox.id] && !checkbox.checked;
+const closeSidebarSections = (targetItem?: HTMLElement): void => {
+    const sections = [...document.querySelectorAll('.js-close-nav-list')];
 
-const closeAllOtherPrimaryLists = (targetItem?: HTMLElement): void => {
-    menuItems.forEach(item => {
-        if (item !== targetItem) {
-            item.removeAttribute('open');
+    sections.forEach(section => {
+        if (section !== targetItem) {
+            section.removeAttribute('open');
         }
     });
 };
 
-const removeOrderingFromLists = (): void => {
-    const mainListItems = qwery('.js-navigation-item');
+const toggleSidebar = (trigger: HTMLElement): void => {
+    const documentElement = document.documentElement;
+    const sidebarToggle = document.querySelector('.js-change-link');
+    const openClass = 'new-header__nav__menu-button--open';
+    const globalOpenClass = 'nav-is-open';
+    const isOpen = trigger.getAttribute('aria-expanded') === 'true';
 
-    mainListItems.forEach(item => {
-        const listItem = item;
+    if (!sidebar || !sidebarToggle) {
```

Comment:
If you passed `sidebar` into the `toggleSidebar` function as a [`Maybe`](https://flow.org/en/docs/types/maybe/) type
```
const toggleSidebar = (trigger: HTMLElement, sidebar: ?Element): void => {
...
}
```
Then flow will check at compile time that you have done a null check against sidebar in the body of the function. 
As we are refactoring a lot of code to es6 we should be using the type system to it's fullest. It's also a more functional approach to pass in all external dependencies into a function.
Sorry for the drive by comment!